src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -3605,13 +3605,14 @@ internal EffectiveParameters(ImmutableArray<TypeWithAnnotations> types, Immutabl
 
             if (arguments.IsExtensionMethodInvocation)
             {
-                var inferredFromFirstArgument = MethodTypeInferrer.InferTypeArgumentsFromFirstArgument(
+                var canInfer = MethodTypeInferrer.CanInferTypeArgumentsFromFirstArgument(
                     _binder.Compilation,
                     _binder.Conversions,
                     method,
                     args,
-                    useSiteInfo: ref useSiteInfo);
-                if (inferredFromFirstArgument.IsDefault)
+                    useSiteInfo: ref useSiteInfo,
+                    out _);
+                if (!canInfer)
```

Comment:
This code path only needs a boolean answer. Previously, `InferTypeArgumentsFromFirstArgument` could allocate, and the allocated array is just thrown away. I'm not sure though if the new method name is good.

Reply:
This change moved to #69071